mllib/src/main/scala/org/apache/spark/ml/tree/impl/RandomForest.scala

```diff
@@ -1107,9 +1108,11 @@ private[spark] object RandomForest extends Logging {
         mutableTreeToNodeToIndexInfo
           .getOrElseUpdate(treeIndex, new mutable.HashMap[Int, NodeIndexInfo]())(node.id)
           = new NodeIndexInfo(numNodesInGroup, featureSubset)
+        numNodesInGroup += 1
+        memUsage += nodeMemUsage
+      } else {
+        flag = false
```

Comment:
We cannot clear stack here, because the nodes in the stack are used for all iteration. 
Native scala doesn't support break, we have to import other package to use break, seems not necessary. 
We cannot use nodeStack.pop at the top, because we don't know whether it can be grouped here.
How about change "flag" to "groupDone" means Group is done, don't need to loop again.